### PR TITLE
transaction.scm unit testing

### PR DESCRIPTION
**Pull request by christopherlam**
and TR enhancement: "Display / Summary Table", new feature with test available.

This builds upon #288 and implements the summary table generation. Also is a continuation of #270.

### CODE REVIEW

File: gnucash/report/standard-reports/test/test-transaction.scm [hunk 1]
```diff
@@ -0,0 +1,864 @@
...
+      (let* ((sxml (options->sxml options "sign-reversal is credit-accounts, correct signs of amounts?"))) ;out-28.html
+        (test-equal "sign-reversal is credit-accounts, correct signs of amounts"
+          '(#f #t #t #f #f #t #t #t #t #t #t #t #t #f)
+          (map (lambda (s) (not (string-contains s "-")))
+               ((sxpath '(// (table 1) // tr // (td -1) // a // *text*)) sxml))))
+
+      ;; test debit/credit dual columns
+      (set! options (default-testing-options))
+      (set-option! options "Display" "Amount" 'double)
+      (let* ((sxml (options->sxml options "dual columns"))) ;out-29.html
```
Comment (christopherlam) [inline gnucash/report/standard-reports/test/test-transaction.scm]: oops forgot to write this test

**Comment by christopherlam** [comment]
> This branch is currently an unclean mix of PR#288 and some new commits. While it can be merged without conflicts it makes for some ugly commit history.
> Can you please rebase it to current unstable so the already merged commits get pruned out ? Thanks

Actually, rebasing on unstable doesn't change anything; it's apparently based on both... If #288 is merged as it stands, the common commits should disappear. If not, I'll rebase this branch on new unstable and repush.

image

**Comment by gjanssens** [comment]
My apologies. I meant for you to rebase **after** I pushed PR#288, but forgot to do the latter...
Can you retry ?

**Comment by christopherlam** [comment]
Rebased. Funny that everytime I look at code I see a new forgotten edge case.

**Comment by gjanssens** [comment]
Thanks!

**Pull request merged**